specs/simple-serialize.md

```diff
@@ -1,424 +1,108 @@
-| `LENGTH_BYTES`    |   4   | Number of bytes used for the length added before a variable-length serialized object. |
-| `SSZ_CHUNK_SIZE`  |  128  | Number of bytes for the chunk size of the Merkle tree leaf.                           |
+* `byte` to `uint8`
+* `bytes` to `byte[]`
+* `bytesN` to `byte[N]`
 
-## Overview
+## Serialization
 
-### Serialize/Encode
+We reccursively define the `serialize` function which consumes an object `value` (of the type specified) and returns a byte string of type `bytes`.
 
-#### uintN
-
-| uint Type | Usage                                                      |
-|:---------:|:-----------------------------------------------------------|
-| `uintN`   | Type of `N` bits unsigned integer, where ``N % 8 == 0``.   |
-
-Convert directly to bytes the size of the int. (e.g. ``uint16 = 2 bytes``)
-
-All integers are serialized as **little endian**.
-
-| Check to perform       | Code                  |
-|:-----------------------|:----------------------|
-| Size is a byte integer | ``int_size % 8 == 0`` |
+#### `uintN`
 
 ```python
-assert(int_size % 8 == 0)
-buffer_size = int_size / 8
-return value.to_bytes(buffer_size, 'little')
+assert N in [8, 16, 32, 64, 128, 256]
+return value.to_bytes(N // 8, 'little')
 ```
 
-#### bool
-
-Convert directly to a single 0x00 or 0x01 byte.
-
-| Check to perform  | Code                       |
-|:------------------|:---------------------------|
-| Value is boolean  | ``value in (True, False)`` |
+#### `bool`
 
 ```python
-assert(value in (True, False))
+assert value in (True, False)
 return b'\x01' if value is True else b'\x00'
 ```
 
-#### bytesN
-
-A fixed-size byte array.
-
-| Checks to perform                      | Code                 |
-|:---------------------------------------|:---------------------|
-| Length in bytes is correct for `bytesN` | ``len(value) == N`` |
-
-```python
-assert(len(value) == N)
-
-return value
-```
-
-#### List/Vectors
-
-Lists are a collection of elements of the same homogeneous type.
-
-| Check to perform                            | Code                        |
-|:--------------------------------------------|:----------------------------|
-| Length of serialized list fits into 4 bytes | ``len(serialized) < 2**32`` |
-
-1. Serialize all list elements individually and concatenate them.
-2. Prefix the concatenation with its length encoded as a `4-byte` **little-endian** unsigned integer.
-
-We define `bytes` to be a synonym of `List[bytes1]`.
-
-**Example in Python**
+#### Containers, tuples, lists
 
 ```python
-
-serialized_list_string = b''
-
-for item in value:
-   serialized_list_string += serialize(item)
-
-assert(len(serialized_list_string) < 2**32)
-
-serialized_len = (len(serialized_list_string).to_bytes(LENGTH_BYTES, 'little'))
-
-return serialized_len + serialized_list_string
+serialized_bytes = ''.join([serialize(element) for element in value])
+LENGTH_BYTES = 4
```

Comment:
I think it clearer to keep the constant section at the top and define `LENGTH_BYTES` and `CHUNK_SIZE` there. And then in `Merkleization` section to call them "`CHUNK_SIZE`-byte chunks"